app/components/TimeUntilUnveil.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from "react";
+
+const getOrdinalSuffix = (day: number): string => {
+  const suffixes = ["th", "st", "nd", "rd"];
+  const value = day % 100;
+  return suffixes[(value - 20) % 10] || suffixes[value] || suffixes[0];
+};
+const formatUnveilDate = (date: Date): string => {
+  const day = date.getDate();
+  const month = date.toLocaleString("default", { month: "long" });
+  const year = date.getFullYear();
+  const hours = date.getHours();
+  const minutes = date.getMinutes();
+
+  const ordinalDay = `${day}${getOrdinalSuffix(day)}`;
+
+  const formattedTime = `${hours}:${minutes < 10 ? `0${minutes}` : minutes}`;
+
+  return `${ordinalDay} ${month} ${year} ${formattedTime}`;
+};
+
+const calculateTimeUntilUnveil = (unveilDate: Date): string => {
+  const now = new Date();
+  const timeDiff = unveilDate.getTime() - now.getTime();
+
+  if (timeDiff <= 0) {
+    return "Unveiled";
+  }
+
+  const days = Math.floor(timeDiff / (1000 * 3600 * 24));
+  const hours = Math.floor((timeDiff % (1000 * 3600 * 24)) / (1000 * 3600));
+  const minutes = Math.floor((timeDiff % (1000 * 3600)) / (1000 * 60));
+  const seconds = Math.floor((timeDiff % (1000 * 60)) / 1000);
+
+  const formattedHours = hours < 10 ? `0${hours}` : hours;
+  const formattedMinutes = minutes < 10 ? `0${minutes}` : minutes;
+  const formattedSeconds = seconds < 10 ? `0${seconds}` : seconds;
+
+  return `${days} Days - ${formattedHours}hr:${formattedMinutes}min:${formattedSeconds}sec`;
+};
+
+interface TimeUntilUnveilProps {
+  unveilDate: Date;
+}
+
+const TimeUntilUnveil: React.FC<TimeUntilUnveilProps> = ({ unveilDate }) => {
+  const [timeRemaining, setTimeRemaining] = useState<string>(
+    calculateTimeUntilUnveil(unveilDate),
+  );
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      const newTimeRemaining = calculateTimeUntilUnveil(unveilDate);
+      setTimeRemaining(newTimeRemaining);
+      if (newTimeRemaining === "Unveiled") {
+        clearInterval(interval);
+      }
+    }, 1000);
+    return () => clearInterval(interval);
+  }, [unveilDate]);
```

Comment:
remove this, we want the function to be called only once when the page  is loaded (which has alraeady been calculated and passed as a value) and not every second.